patch_files/patches/46_resume_cancelled_cloudify_workflows_workflow_context.py

```diff
@@ -0,0 +1,37 @@
+diff --git a/cloudify/workflows/workflow_context.py b/cloudify/workflows/workflow_context.py
+index af9adf9..6a05d18 100644
+--- a/cloudify/workflows/workflow_context.py
++++ b/cloudify/workflows/workflow_context.py
+@@ -1270,14 +1270,13 @@ class _TaskDispatcher(object):
+         ping_handler = amqp_client.BlockingRequestResponseHandler(
+             exchange=task['target'])
+         client = self._get_client(task)
+-        client.add_handler(handler)
+         client.add_handler(ping_handler)
++        client.add_handler(handler)
+         task.update({
+             'client': client,
+             'handler': handler,
+             'ping_handler': ping_handler
+         })
+-        client.consume_in_thread()
```

Comment:
this change is not in master, but will be:
moving `consume_in_thread` to AFTER the callback has been added fixes a race condition between receiving the response message, and adding a callback that will handle it